log/sockstatlog/logger.go

```diff
@@ -114,11 +114,24 @@ func NewLogger(logdir string, logf logger.Logf, logID logid.PublicID) (*Logger,
 		HTTPC: &http.Client{Transport: logger.tr},
 	}, logf)
 
-	go logger.poll()
-
 	return logger, nil
 }
 
+// SetLoggingEnabled enables or disables logging.
+// When disabled, socket stats are not polled and no new logs are written to disk.
+// Existing logs can still be fetched via the C2N API.
+func (l *Logger) SetLoggingEnabled(v bool) {
+	old := l.enabled.Load()
+	if l.enabled.CompareAndSwap(old, v) {
```

Comment:
Seems like you should also check that `v != old`, otherwise you'll start another poll if someone calls `SetLoggingEnabled` twice.

Reply:
oops, never mind :)  forget to reset old on each of those tests

Reply:
done